pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -85,9 +124,10 @@ in stdenv.mkDerivation {
       pcre2 ] ++
       stdenv.lib.optional withKexectools kexectools ++
       stdenv.lib.optional withLibseccomp libseccomp ++
-    [ libffi audit lz4 bzip2 libapparmor
-      iptables gnu-efi
-    ] ++ stdenv.lib.optional withSelinux libselinux;
+      [ libffi audit lz4 bzip2 libapparmor iptables ] ++
+      stdenv.lib.optional withEfi gnu-efi ++
+      stdenv.lib.optional withSelinux libselinux ++
+      stdenv.lib.optionals withHomed [ cryptsetup.dev p11-kit libfido2 ];
```

Comment:
We don't want cryptsetup only when homed is enabled - This is also generally useful to have systemd understand (and act upon) `/etc/crypttab` on - see https://github.com/NixOS/nixpkgs/pull/66856

Maybe adding another `withCryptsetup` flag that sets `-Dcryptsetup=true`, and is a requirement for `withHomed`?

Looking at https://github.com/NixOS/nixpkgs/pull/97008/files#diff-036410e9211b4336186fc613f7200b12R18035, it seems to be fine if this would be set to `false` for `systemdMinimal`.